### DISCUSSION TITLE
iRobot brand support?

### DISCUSSION THREAD

**Discussion by QuantumToasted** [Hypfer/Valetudo#1671] [Q&A / Support] [answered]
Hi there, I could not find any closed issues, discussions, or anything related to this, so apologies if this has been answered or explained somewhere.

My family currently owns an iRobot Roomba 690, and ever since we have upgraded our networking and APs it has just straight up decided to not work with the standard software anymore (so we cannot schedule it, control it remotely, etc). It still functions perfectly (and will run if you just push its usual "Clean" button) and we really wanted to avoid purchasing an entirely new robot vacuum just because the software was being shitty. I ended up stumbling upon Valetudo and couldn't find anything listed in the realm of support for the brand - is it due to the vacuums being too proprietary/locked-down and pretty much impossible to root or gain control over? If there is anything that I can do to help as the owner of one, let me know.

Thanks in advance!

**Comment by Hypfer** [comment, marked as answer]
Conceptually, Valetudo is built around robots that report some kind of live map of their surroundings.
That alone rules out iRobot support entirely.

Apart from that, Valetudo also runs on the robot, meaning that
- You need a way to run things on your robot
- The robots hardware needs to be capable of running Valetudo

while you could run Valetudo on a different host, not only would that be a rather mediocre experience compared to the self-contained vacuum appliance with a Webinterface, but you'd need to be able to 
- somehow get the robot to talk to your backend instead of the cloud
- understand how exactly that protocol works


So TL;DR: Not possible. Not even close to being possible.

It might be possible to run that Roomba 690 cloudless, however certainly not with Valetudo as it's basically a whole different class of vacuum robot.
You could call it legacy I suppose.

**Reply by QuantumToasted** [reply]
Appreciate the quick and insightful response. I have literally no idea how **any** brand of robot vacuums operate, let alone specifically iRobot, so I went into this question-asking process completely blind. Was just trying to save my family from having to use our current vacuum in its current state of refusing to connect to the internet. (The whole point of spending the money on the "smart" vacuum was to be able to program it so we didn't have to remember to run it, lol)

**Reply by Hypfer** [reply]
While I understand that it would feel wrong to discard a still working robot, I can also strongly recommend upgrading to something more recent and LIDAR-based.
It's a completely different experience to these older vacuum robot offerings. Not only is the navigation **so much better** due to much better sensors but you'll also get a ton of additional features such as being able to place virtual walls or only clean specific rooms.

The current recommendations for buying a Valetudo-supported vacuum robot can be found in the docs:
https://valetudo.cloud/pages/general/buying-supported-robots.html

I'd also recommend just throwing those model names into the youtube search to see what they can do. They really are just something entirely different to that old roomba 690
